src/goose/brokers/redis/batch.clj

```diff
@@ -0,0 +1,27 @@
+(ns goose.brokers.redis.batch
+  (:require
+    [goose.brokers.redis.commands :as redis-cmds]
+    [goose.brokers.redis.retry :as retry]
+    [goose.defaults :as d]))
+
+(defn metadata-update [next]
+  (fn [{:keys [redis-conn]
+        :as   opts}
+       {:keys [id, batch-id]} job]
+    (if (contains? job :batch-id)
+      (try
+        (let [response (next opts job)
+              _ (redis-cmds/del-and-add redis-conn
+                                        (d/batch-enqueued-set batch-id)
```

Comment:
Have we considered a "retrying-set"?
Will complicate the middleware a bit, but will add 1 more important stage to a job. i.e. it isn't in enqueued state forever, rather retrying because it failed.

Reply:
Yes, I am considering this. More conversations happening in slack, to consider tradeoffs between having the retry-set.